docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1260 @@
+
+#### 4.3.5. List Feature
+
+##### 4.3.5.1. Implementation
+
+The following sequence diagram shows how the list operation in the notebook mode works:
+
+![Sequence Diagram for List](diagrams/class/jpeg/SequenceDiagram_ListSection.jpg)
+
+Given below is an example usage scenario and how the list function behaves.
+
+Step 1. The user types `list /s`. The `list /s` command is passed through
+`InputParser#getCommandFromInput`, which constructs a `ListCommandNoteMode` object and calls
+`ListCommandNotebookMode#execute()`.
+
+Step 2. `execute()` is called, which then calls the print functions based on the `AppMode` field in `AppState`.
+
+Step 3. The program prints the contents corresponding to the input or throw an exception if the command is
+invalid.
+
+##### 4.3.5.2. Design Considerations
+
+### 4.4. Storage (Neil)
+
+The `Storage` class reads and writes the application state to and from a text file.
+
+#### 4.4.1. Storage format
+
+This section describes the format used to store the TaskList and NotebookShelf from the application state in a plain
+text file.
+
+:exclamation: The operating system newline character is used to terminate lines; that is, `\r\n` on Windows and
+`\n` on UNIX-based systems.
+
+##### 4.4.1.1. TaskList
+
+For `TaskList`, the format is as follows:
+
+* One line containing an integer `n`, the number of tasks.
+* `3n` lines follow, every 3 lines describing one task.
+* The first line contains the task title as a String.
+* The second line contains the task due date, in the format `dd-MM-yyyy hhmm`.
+* The third line contains a value `true` or `false`, indicating whether the task is done. (`true` means done,
+`false` means not done.)
+
+##### 4.4.1.2. Page
+
+For a single page, the format is as follows:
+
+* One line containing the title of the page.
+* One line containing the content of the page. Newline characters in the content are replaced with the string `"~~~"`.
+
+##### 4.4.1.3. Section
+
+For a single section, the format is as follows:
+
+* One line containing an integer `p`, the number of pages in the section.
+* `p` pages are then described, as specified [here](#4412-page).
+
+##### 4.4.1.4. Notebook
+
+For a single notebook, the format is as follows:
+
+* One line containing an integer `s`, the number of sections.
+* `s` sections are then described, as specified [here](#4413-section)
+
+##### 4.4.1.5. NotebookShelf
+
+For `NotebookShelf`, the format is as follows:
+
+* One line containing an integer `n`, the number of notebooks.
+* `n` notebooks are then described, as specified [here](#4414-notebook).
+
+#### 4.4.2 Implementation
+
+`TaskList`, `Task`,`NotebookShelf`, `Notebook`, `Section`, and `page` contain methods called `serialize()`.
+These methods return`String` representations of themselves, as specified [here](#441-storage-format).  
+
+The `serialize()` method in `TaskList` calls the `serialize()` methods for every `Task` object within
```

Comment:
1 suggestion - an example of how it serializes might  be a bit more informative?